pageserver/src/layered_repository/image_layer.rs

```diff
@@ -241,6 +242,22 @@ impl ImageLayer {
         }
     }
 
+    fn temp_path_for(
+        conf: &PageServerConf,
+        timelineid: ZTimelineId,
+        tenantid: ZTenantId,
+        fname: &ImageFileName,
+    ) -> PathBuf {
+        let rand_string: String = rand::thread_rng()
+            .sample_iter(&Alphanumeric)
+            .take(8)
+            .map(char::from)
+            .collect();
+
+        conf.timeline_path(&timelineid, &tenantid)
+            .join(format!("{}.temp.{}", fname, rand_string))
```

Comment:
Same here

Reply:
Fair enough.
While on it, I also implemented the file cleanup to close #1732